test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.client.cs

```diff
@@ -212,72 +215,217 @@ public void GrpcAndHttpClientInstrumentationWithSuppressInstrumentation(bool sho
         }
 
         [Fact]
-        public void GrpcPropagatesContextWithSuppressInstrumentation()
+        public void GrpcPropagatesContextWithSuppressInstrumentationOptionSetToTrue()
         {
-            var uri = new Uri($"http://localhost:{this.server.Port}");
-            var processor = new Mock<BaseProcessor<Activity>>();
+            try
+            {
+                var uri = new Uri($"http://localhost:{this.server.Port}");
+                var processor = new Mock<BaseProcessor<Activity>>();
 
-            using var source = new ActivitySource("test-source");
+                using var source = new ActivitySource("test-source");
 
-            using (Sdk.CreateTracerProviderBuilder()
-                .AddSource("test-source")
-                .AddGrpcClientInstrumentation(o =>
-                {
-                    o.SuppressDownstreamInstrumentation = true;
-                })
-                .AddHttpClientInstrumentation()
-                .AddAspNetCoreInstrumentation(options =>
+                var propagator = new Mock<TextMapPropagator>();
+                propagator.Setup(m => m.Inject<HttpRequestMessage>(It.IsAny<PropagationContext>(), It.IsAny<HttpRequestMessage>(), It.IsAny<Action<HttpRequestMessage, string, string>>()))
+                    .Callback<PropagationContext, HttpRequestMessage, Action<HttpRequestMessage, string, string>>((context, message, action) =>
+                    {
+                        action(message, "customField", "customValue");
+                    });
+
+                Sdk.SetDefaultTextMapPropagator(new CompositeTextMapPropagator(new TextMapPropagator[]
                 {
-                    options.Enrich = (activity, eventName, obj) =>
+                    new TraceContextPropagator(),
+                    propagator.Object,
+                }));
+
+                using (Sdk.CreateTracerProviderBuilder()
+                    .AddSource("test-source")
+                    .AddGrpcClientInstrumentation(o =>
                     {
-                        switch (eventName)
+                        o.SuppressDownstreamInstrumentation = true;
+                    })
+                    .AddHttpClientInstrumentation()
+                    .AddAspNetCoreInstrumentation(options =>
+                    {
+                        options.Enrich = (activity, eventName, obj) =>
```

Comment:
Don't think Enrich is relevant to this test. I'd suggest to remove it from this particular test